r/tests/testthat/test-Array.R

```diff
@@ -1321,3 +1321,19 @@ test_that("Array to C-interface", {
   delete_arrow_schema(schema_ptr)
   delete_arrow_array(array_ptr)
 })
+
+test_that("direct creation of Decimal Arrays (ARROW-11631)", {
```

Comment:
I'm a little confused.  I expected to see an R array of doubles get converted to an Arrow array.  I didn't expect to see decimals get created directly.

Reply:
Well, in python it would be something like:

```
# create an array of python decimals
x = [decimal.Decimal('12.34'), decimal.Decimal('23.45')]
# convert to an array of arrow decimals
arr = pa.array(x)
# arr
# <pyarrow.lib.Decimal128Array object at 0x7fd6a40dd120>
# [
#   12.34,
#   23.45
# ]
```

It looks like there are tests like this in `test-Array.R`:

```
test_that("Integer Array", {
  ints <- c(1:10, 1:10, 1:5)
  x <- expect_array_roundtrip(ints, int32())
})
```

Also in`test-Array.R` I see something like...

```
  expect_error(
    Array$create(as.double(1:10), type = decimal(4, 2)),
    "You might want to try casting manually"
  )
```

Shouldn't this work now?

Reply:
True, you can't do proper round trip if there is no native decimals.  I see now my confusion.  I read `Array$create(1, type = decimal128(10, 2))` too quickly and thought it was creating an Arrow scalar from a single R value.  I didn't realize it was treating `1` as an array of size 1.